### PR TITLE
Identify & Replace Keypress

### LINKED ISSUE #123: Add Ctrl-F4 key combination to replace current found selection and then find next
- Ctrl-F4 - Perform replace and find next
  - Just like F4, but after the replace, moves to the next found selection (ie: it's like pressing F4 followed by F3)

### LINKED ISSUE #157: Identify - Add key commands
Add shortcut keys `F6` and `Ctrl-I` to display the Identify dialog (if not already displayed) and performs the Identification process on all available ports, showing the results in the dialog.  Pressing the key command again while dialog is open causes the list to refresh.

Split out from #128

### PR DESCRIPTION
**Pull request by spatializes**
#### What's this PR do?

Adds keypress functionality for identify and replace. Also refactors dependent functionality into Flux.
#### Where should the reviewer start?

Pull this branch.
npm run build
#### How should this be manually tested?

Load a file with some text. Press CTRL-F4 to test the CodeMirror replace functionality. Press enter to replace as matches are found.
NOTE: for OSX users - CTRL-F4 is used by the OS to move window focus. To test this, you will need to disable the OSX keyboard shortcut. System Preferences > Keyboard > Shortcuts > Keyboard.

Press CTRL-I or F6 to trigger the download modal to pop up. Try downloading code to the STAMP. Try to hit Cancel and the ESC key in the download modal.

Test deleting files and creating files and saving. Ensure delete functionality is working as it was also moved to Flux.
#### Any background context you want to provide?

NOTE: Still need to test the download functionality with a STAMP. It along with everything else should be working.

To enable the shortcut for identify, I moved the download overlay into sidebar index.js with the other overlay functions. I also cleaned up the remaining overlay in file-operations.js, DeleteOverlay, by moving it into index.js. These might belong somewhere else but I thought I would follow the pattern we started. They are now taking advantage of Flux by using state to show and hide. It should be very easy to move these if we want a view file that holds all of the overlays, or something like that. In addition, I am passing the actions down as props into file-operations.js instead of requiring the actions in the component. I cleaned up any remaining actions in file-operations to follow this pattern as well, removing all actions from the component.
#### What are the relevant tickets?

Closes #157
Closes #123

### CODE REVIEW

**Comment by PropGit** [comment]
Thanks @drquinn !

File: src/actions/system.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ class SystemActions {
     this.dispatch();
   }
 
+  identify() {
```
Comment (phated) [inline src/actions/system.js]: This is a device action

File: src/actions/file.js [hunk 2]
```diff
@@ -15,10 +15,22 @@ class FileActions {
     this.dispatch();
   }
 
+  handleError(err) {
+    this.dispatch(err);
+  }
+
+  handleSuccess(err) {
```
Comment (phated) [inline src/actions/file.js]: not an `err`

File: plugins/sidebar/file-operations.js [hunk 3]
```diff
@@ -5,83 +5,21 @@ const { Menu, MainButton, ChildButton } = require('react-mfb-iceddev');
...
-        onCancel={hideOverlay}
-        irken={this.props.irken}
-        handleSuccess={this.handleSuccess}
-        handleError={this.handleError} />
-    );
-
-    this.renderOverlay(component);
-  },
   render: function(){
+    const { newFile, processSave, showDelete, showDownload } = this.props;
```
Comment (phated) [inline plugins/sidebar/file-operations.js]: require these instead of passing them

File: plugins/editor/key-extension.js [hunk 4]
```diff
@@ -24,6 +25,20 @@ const keyExtension = {
           findPrevious();
         }
       },
+      identify1: {
+        code: 'F6',
```
Comment (phated) [inline plugins/editor/key-extension.js]: use an array and make your mapping logic smarter

**Pull request merged**